internal/deploy/yaml.go

```diff
@@ -0,0 +1,213 @@
+package deploy
+
+import (
+	"os"
+	"strings"
+
+	"github.com/shipa-corp/ketch/internal/errors"
+	"sigs.k8s.io/yaml"
+
+	ketchv1 "github.com/shipa-corp/ketch/internal/api/v1beta1"
+)
+
+// Application represents the fields in an application.yaml file that will be
+// transitioned to a ChangeSet.
+type Application struct {
+	Version        *string    `json:"version"`
+	Type           *string    `json:"type"`
+	Name           *string    `json:"name"`
+	Image          *string    `json:"image"`
+	Framework      *string    `json:"framework"`
+	Description    *string    `json:"description"`
+	Environment    *[]string  `json:"environment"`
+	RegistrySecret *string    `json:"registrySecret"`
+	Builder        *string    `json:"builder"`
+	BuildPacks     *[]string  `json:"buildPacks"`
+	Processes      *[]Process `json:"processes"`
+	CName          *CName     `json:"cname"`
+	AppUnit        *int       `json:"appUnit"`
+}
+
+type Process struct {
+	Name  string `json:"name"`  // required
+	Cmd   string `json:"cmd"`   // required
+	Units *int   `json:"units"` // unset? get from AppUnit
+	Ports []Port `json:"ports"` // appDeploymentSpec
+	Hooks Hooks  `json:"hooks"`
+}
+
+type Port struct {
+	Protocol   string `json:"protocol"`
+	Port       int    `json:"port"`
+	TargetPort int    `json:"targetPort"`
+}
+
+type Hooks struct {
+	Restart Restart `json:"restart"`
+}
+
+type Restart struct {
+	Before string `json:"before"`
+	After  string `json:"after"`
+}
+
+type CName struct {
+	DNSName string `json:"dnsName"`
+}
+
+var (
+	defaultVersion  = "v1"
```

Comment:
const?

Reply:
I used vars because can't take the address of consts. Preference for using `StrPtr()` function?

Reply:
I would vote for safety. 
Also, there is `k8s.io/utils/pointer` package with `StringPtr` function, we can use it

Reply:
Switched to const and moved our pointer utils to `utils`. k8s.io/utils/pointer didn't have a plain IntPtr. Shame!